packages/truffle-core/lib/mnemonics/mnemonic.js

```diff
@@ -0,0 +1,74 @@
+/**
+* @module mnemonic;
+* @requires module:Configstore
+* @requires module:random
+* @requires module:bip39
+* @requires module:ethereumjs-wallet/hdkey
+*/
+
+const Configstore = require('configstore');
+const random = require('./random');
+const seedrandom = require('seedrandom');
+const bip39 = require('bip39'); 
+const hdkey = require('ethereumjs-wallet/hdkey');
+const crypto = require('crypto');
+
+const mnemonic = {
+
+  /**
+  * gets user-level mnemonic from user config, and if missing generates a new mnemonic
+  * @returns {String} mnemonic
+  */
+  mnemonic: function () {
+    let mnemonic;
+    
+    const defaultUserConfig = new Configstore('truffle', {}, { globalConfigPath: true });
+    const userMnemonicExists = defaultUserConfig.get("mnemonic");
+    if(!userMnemonicExists) {
+      mnemonic = bip39.entropyToMnemonic(crypto.randomBytes(16).toString('hex'));
```

Comment:
I copied this way of generating a mnemonic from ethagen (https://github.com/benjamincburns/ethagen/blob/master/wallet.js), and read up on crypto.randomBytes in the process. It should be cryptographically secure enough from what I read but I wanted to point it out in case I'm wrong about that.